### PR TITLE
Respawn walls at MMO equivalent distance

### PR DESCRIPTION
**Pull request by shibdib**
move walls to mmo equivalent (still needs edge cases checked to make sure both sides have exits)

I have no way to test this (I don't know if I got the corners mixed up at 646 - 653 because it was making my head hurt to figure out what direction was up)

### CODE REVIEW

File: public/renderer.js [hunk 1]
```diff
@@ -557,31 +557,67 @@ function getSectorBoundsAllBus (roomInSector) {
...
+        x2 = 20
+        y2 = 30 + i
+        break
+    }
+    if (x !== undefined && y !== undefined && x < 50 && y < 50) {
+      room.objects.push({type: 'constructedWall', room: room.room, x, y, decayTime: {timestamp: decayTime}})
+    }
+    if (x2 !== undefined && y2 !== undefined && x2 < 50 && y2 < 50) {
+      room.objects.push({type: 'constructedWall', room: room.room, x2, y2, decayTime: {timestamp: decayTime}})
+    }
```
Comment (psy3727) [inline public/renderer.js:619]: //fill in missing hole
    if (borderSide === 'bottomRight') {
      room.objects.push({type: 'constructedWall', room: room.room, x: 20, y: 20, decayTime: {timestamp: decayTime}})
    }

File: public/renderer.js [hunk 2]
```diff
@@ -607,7 +643,15 @@ function makeRespawnSector (roomInSector, openTime, decayTime) {
 
       if (horx % 10 == 0 || very % 10 == 0) {
         room.bus = true
-        if (x == start.x && y > start.y && y < (start.y + 10)) {
+        if (x == start.x && y == start.y) {
+          makeRespawnSectorWall(room, 'bottomRight', decayTime)
```
Comment (psy3727) [inline public/renderer.js]: this should be 'topLeft'

File: public/renderer.js [hunk 3]
```diff
@@ -607,7 +643,15 @@ function makeRespawnSector (roomInSector, openTime, decayTime) {
...
+        } else if (x == start.x && y == (start.y + 10)) {
+          makeRespawnSectorWall(room, 'topRight', decayTime)
```
Comment (psy3727) [inline public/renderer.js]: this should be 'bottomLeft'

File: public/renderer.js [hunk 4]
```diff
@@ -607,7 +643,15 @@ function makeRespawnSector (roomInSector, openTime, decayTime) {
...
+        } else if (y == start.y && x == (start.x + 10)) {
+          makeRespawnSectorWall(room, 'bottomLeft', decayTime)
```
Comment (psy3727) [inline public/renderer.js:652]: this should be 'topRight'

File: public/renderer.js [hunk 5]
```diff
@@ -607,7 +643,15 @@ function makeRespawnSector (roomInSector, openTime, decayTime) {
...
+        } else if (y == (start.y + 10) && x == (start.x + 10)) {
+          makeRespawnSectorWall(room, 'topLeft', decayTime)
```
Comment (psy3727) [inline public/renderer.js]: this should be 'bottomRight'

**Review by psy3727 — CHANGES_REQUESTED** [review]
Tested and found a couple errors. With these minor updates it should be good to go.

File: public/renderer.js [hunk 6]
```diff
@@ -523,113 +534,157 @@ function getSectorBoundsAllBus (roomInSector) {
...
+        y = 30
+        x2 = 20
+        y2 = 30 + i
+        break
+    }
+    if (x !== undefined && y !== undefined && x < 50 && y < 50) {
+      room.objects.push({type: 'constructedWall', room: room.room, x, y, decayTime: {timestamp: decayTime}})
+    }
+    if (x2 !== undefined && y2 !== undefined && x2 < 50 && y2 < 50) {
+      room.objects.push({type: 'constructedWall', room: room.room, x2, y2, decayTime: {timestamp: decayTime}})
```
Comment (psy3727) [inline public/renderer.js]: assignment error here, should be:
room.objects.push({type: 'constructedWall', room: room.room, x: x2, y: y2, decayTime: {timestamp: decayTime}})

**Review by psy3727 — CHANGES_REQUESTED** [review]
-found one last error

**Review by psy3727 — APPROVED** [review]
